src/types.rs

```diff
@@ -591,18 +591,48 @@ pub enum Primitive {
     U16(u16),
     U32(u32),
     U64(u64),
-    // TODO: better way?
-    U128([u8; 16]),
-    #[cfg(not(target_pointer_width = "32"))]
-    U256([u64; 4]),
-    #[cfg(target_pointer_width = "32")]
-    U256([u32; 8]),
+    U128(u128),
+    U256(U256), // Using a custom struct for U256
     USize(u32),
     Bool(bool),
     Felt252(FieldElement),
     ClassHash(FieldElement),
     ContractAddress(FieldElement),
 }
+#[derive(Clone, Copy, Debug)]
+pub struct U256([u64; 4]); // Simple representation, adjust as needed
```

Comment:
this is not going to work - the struct is not repr(C) so not exposed to cbindgen